examples/api/lib/widgets/pop_scope/pop_scope.1.dart

```diff
@@ -0,0 +1,250 @@
+  _Tab _tab = _Tab.home;
+
+  final GlobalKey _tabHomeKey = GlobalKey();
+  final GlobalKey _tabOneKey = GlobalKey();
+  final GlobalKey _tabTwoKey = GlobalKey();
+
+  List<_TabPage> _tabHomePages = <_TabPage>[_TabPage.home];
+  List<_TabPage> _tabOnePages = <_TabPage>[_TabPage.home];
+  List<_TabPage> _tabTwoPages = <_TabPage>[_TabPage.home];
+
+  BottomNavigationBarItem _itemForPage(_Tab page) {
+    switch (page) {
+      case _Tab.home:
+        return const BottomNavigationBarItem(
+          icon: Icon(Icons.home),
+          label: 'Home',
+        );
+      case _Tab.one:
+        return const BottomNavigationBarItem(
+          icon: Icon(Icons.one_k),
+          label: 'One',
+        );
+      case _Tab.two:
+        return const BottomNavigationBarItem(
+          icon: Icon(Icons.two_k),
+          label: 'Two',
+        );
+    }
+  }
+
+  Widget _getPage(_Tab page) {
+    switch (page) {
+      case _Tab.home:
+        return _BottomNavTab(
+          key: _tabHomeKey,
+          title: 'Home',
+          color: Colors.brown,
+          pages: _tabHomePages,
+          onChangedPages: (List<_TabPage> pages) {
+            setState(() {
+              _tabHomePages = pages;
+            });
+          },
+        );
+      case _Tab.one:
+        return _BottomNavTab(
+          key: _tabOneKey,
+          title: 'One',
+          color: Colors.deepPurple,
+          pages: _tabOnePages,
+          onChangedPages: (List<_TabPage> pages) {
+            setState(() {
+              _tabOnePages = pages;
+            });
+          },
+        );
+      case _Tab.two:
+        return _BottomNavTab(
+          key: _tabTwoKey,
+          title: 'Two',
+          color: Colors.blueGrey,
+          pages: _tabTwoPages,
+          onChangedPages: (List<_TabPage> pages) {
+            setState(() {
+              _tabTwoPages = pages;
+            });
+          },
+        );
+    }
+  }
+
+  void _onItemTapped(int index) {
+    setState(() {
+      _tab = _Tab.values.elementAt(index);
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Center(
+        child: _getPage(_tab),
+      ),
+      bottomNavigationBar: BottomNavigationBar(
+        items: _Tab.values.map(_itemForPage).toList(),
+        currentIndex: _Tab.values.indexOf(_tab),
+        selectedItemColor: Colors.amber[800],
+        onTap: _onItemTapped,
+      ),
+    );
+  }
+}
+
+class _BottomNavTab extends StatelessWidget {
+  const _BottomNavTab({
+    super.key,
+    required this.color,
+    required this.onChangedPages,
+    required this.pages,
+    required this.title,
+  });
+
+  final Color color;
+  final _TabPageCallback onChangedPages;
+  final List<_TabPage> pages;
+  final String title;
+
+  bool get _popEnabled => pages.length <= 1;
+
+  @override
+  Widget build(BuildContext context) {
+    return PopScope(
+      popEnabled: _popEnabled,
+      onPopped: (bool success) {
+        if (success) {
```

Comment:
I have been struggling to find a use case where people want to handle something when success is true..